src/qt/optionsmodel.cpp

```diff
@@ -71,9 +72,19 @@ void OptionsModel::Init(bool resetSettings)
     fMinimizeOnClose = settings.value("fMinimizeOnClose").toBool();
 
     // Display
-    if (!settings.contains("nDisplayUnit"))
-        settings.setValue("nDisplayUnit", BitcoinUnits::BTC);
-    nDisplayUnit = settings.value("nDisplayUnit").toInt();
+    if (!settings.contains("DisplayBitcoinUnit")) {
+        settings.setValue("DisplayBitcoinUnit", QVariant::fromValue(BitcoinUnit::BTC));
+    }
+    QVariant unit = settings.value("DisplayBitcoinUnit");
+    if (unit.canConvert<BitcoinUnit>()) {
+        m_display_bitcoin_unit = unit.value<BitcoinUnit>();
+    } else {
+        m_display_bitcoin_unit = BitcoinUnit::BTC;
+
+        // If for whatever reason we have an invalid DisplayBitcoinUnit setting,
```

Comment:
9b018ca26db5c9ea4d3abdb432a9016607a5c6b8

I think this comment is unnecessary. Besides, this runs on the first app execution.

Reply:
okies, done.